ReactiveCocoaTests/InterceptingSpec.swift

```diff
@@ -930,4 +990,178 @@ private class InterceptedObject: NSObject {
 	@objc dynamic func testNumericValues(c: CChar, s: CShort, i: CInt, l: CLong, ll: CLongLong, uc: CUnsignedChar, us: CUnsignedShort, ui: CUnsignedInt, ul: CUnsignedLong, ull: CUnsignedLongLong, f: CFloat, d: CDouble, b: CBool) {}
 	@objc dynamic func testReferences(nonnull: NSObject, nullable: NSObject?, iuo: NSObject!, class: AnyClass, nullableClass: AnyClass?, iuoClass: AnyClass!) {}
 	@objc dynamic func testBridgedStructs(p: CGPoint, s: CGSize, r: CGRect, a: CGAffineTransform) {}
+	@objc dynamic func testReturnValuesC(arg: CChar) -> CChar { return arg }
+	@objc dynamic func testReturnValuesS(arg: CShort) -> CShort { return arg }
+	@objc dynamic func testReturnValuesI(arg: CInt) -> CInt { return arg }
+	@objc dynamic func testReturnValuesL(arg: CLong) -> CLong { return arg }
+	@objc dynamic func testReturnValuesLL(arg: CLongLong) -> CLongLong { return arg }
+	@objc dynamic func testReturnValuesUC(arg: CUnsignedChar) -> CUnsignedChar { return arg }
+	@objc dynamic func testReturnValuesUS(arg: CUnsignedShort) -> CUnsignedShort { return arg }
+	@objc dynamic func testReturnValuesUI(arg: CUnsignedInt) -> CUnsignedInt { return arg }
+	@objc dynamic func testReturnValuesUL(arg: CUnsignedLong) -> CUnsignedLong { return arg }
+	@objc dynamic func testReturnValuesULL(arg: CUnsignedLongLong) -> CUnsignedLongLong { return arg }
+	@objc dynamic func testReturnValuesF(arg: CFloat) -> CFloat { return arg }
+	@objc dynamic func testReturnValuesD(arg: CDouble) -> CDouble { return arg }
+	@objc dynamic func testReturnValuesB(arg: CBool) -> CBool { return arg }
+	@objc dynamic func testReturnValuesObject(arg: NSObject) -> NSObject { return arg }
+	@objc dynamic func testReturnValuesObjectOptional(arg: NSObject?) -> NSObject? { return arg }
+	@objc dynamic func testReturnValuesClass(arg: AnyClass) -> AnyClass { return arg }
+}
+
+fileprivate struct ReturnValueTest<I: Equatable> {
+	var object: InterceptedObject
+	var selector: Selector {
+		switch I.self {
+		case is CChar.Type:
+			return #selector(InterceptedObject.testReturnValuesC(arg:))
+		case is CShort.Type:
+			return #selector(InterceptedObject.testReturnValuesS(arg:))
+		case is CInt.Type:
+			return #selector(InterceptedObject.testReturnValuesI(arg:))
+		case is CLong.Type:
+			return #selector(InterceptedObject.testReturnValuesL(arg:))
```

Comment:
I think this logic unnecessarily complicates the test. It would be better to pass in the `Selector` directly if possible.